### PR TITLE
HPCC-15194 Log and check connection to dali

### PR DESCRIPTION
**Pull request by wangkx**
We got several reports about files being locked by ESP. Without
logging (and checking) connection ID for each lock, it may be
difficult to find out which ESP call creates a file lock and
whether the lock is a leak or not.

In this fix, an option 'LogConnection' is added to the CClient-
SDSManager. If the option is set, the SDSManager will log the
Connection ID when a connection is created to dali. I also add
the Connection ID back to daliadmin dalilocks report.

Signed-off-by: wangkx kevin.wang@lexisnexis.com

### CODE REVIEW

**Comment by wangkx** [comment]
@jakesmith please review.

File: dali/base/dacsds.cpp [hunk 1]
```diff
@@ -1776,6 +1778,8 @@ IRemoteConnection *CClientSDSManager::connect(const char *xpath, SessionId id, u
             mb.read(connId);
             conn = new CRemoteConnection(*this, connId, xpath, id, mode & ~RTM_CREATE_MASK, timeout);
             assertex(conn);
+            if (queryProperties().getPropBool("Client/@LogConnection"))
```
Comment (jakesmith) [inline dali/base/dacsds.cpp]: minor: a bit of a shame to be checking a IPropertyTree per connection.
I'm surprised there are others actually (in the constructor), e.g. serverIterAvailable - which is static. It should really setup once in a member variable in the manager (CClientSDSManager) and check that.

For this one, which could change, setConfigOpt() could spot and set member variable.

Probably not worth changing for this PR, another PR could move this one and the pre-existing ones (serverIterAvailable, serverIter, useAppendOpt, serverGetIdsAvailable) at the same time.

File: esp/services/ws_smc/ws_smcService.cpp [hunk 2]
```diff
@@ -141,6 +141,7 @@ void CWsSMCEx::init(IPropertyTree *cfg, const char *process, const char *service
 
     xpath.setf("Software/EspProcess[@name=\"%s\"]/EspService[@name=\"%s\"]/ActivityInfoCacheSeconds", process, service);
     activityInfoCacheSeconds = cfg->getPropInt(xpath.str(), DEFAULTACTIVITYINFOCACHETIMEOUTSECOND);
+    querySDS().setConfigOpt("Client/@LogConnection", "1");
```
Comment (jakesmith) [inline esp/services/ws_smc/ws_smcService.cpp]: Should probably be picked up as configuration setting and defaulted to off.
"true" vs "1" would make more sense also.

**Comment by jakesmith** [comment]
@wangkx - looks good, 1 minor issue.

File: esp/services/ws_smc/ws_smcService.cpp (hunk 2 above)
Comment (richardkchapman) [inline esp/services/ws_smc/ws_smcService.cpp]: I doubt we want to default it on - it could be fairly expensive

**Comment by wangkx** [comment]
@jakesmith the code is added to configure the Client/@LogConnection and the default is off now. Please review.

File: initfiles/componentfiles/configxml/espsmcservice.xsd.in [hunk 3]
```diff
@@ -125,10 +125,17 @@
                     </xs:appinfo>
                 </xs:annotation>
             </xs:attribute>
-            <xs:attribute name="disableUppercaseTranslation" type="xs:boolean" use="optional" default="false">
+            <xs:attribute name="enableSystemUseRewrite" type="xs:boolean" use="optional" default="false">
+                <xs:annotation>
+                    <xs:appinfo>
+                        <tooltip>To disable ESP Service links for System Servers that use rewrite rules.</tooltip>
```
Comment (richardkchapman) [inline initfiles/componentfiles/configxml/espsmcservice.xsd.in]: This looks like an unrelated change?

Comment (wangkx) [inline initfiles/componentfiles/configxml/espsmcservice.xsd.in]: Sorry. Something is wrong when I merge my code. I will fix it.

File: dali/base/dacsds.cpp [hunk 4]
```diff
@@ -1730,6 +1730,8 @@ IRemoteConnections *CClientSDSManager::connect(IMultipleConnector *mConnect, Ses
                 mConnect->getConnectionDetails(c, xpath, mode);
                 Owned<CRemoteConnection> conn = new CRemoteConnection(*this, connId, xpath, id, mode & ~RTM_CREATE_MASK, timeout);
                 assertex(conn.get());
+                if (queryProperties().getPropBool("Client/@LogConnection"))
+                    DBGLOG("SDSManager::connect() - IMultipleConnector: RemoteConnection ID<%" I64F "x>,  timeout<%d>", connId, timeout);
```
Comment (jakesmith) [inline dali/base/dacsds.cpp]: very trivial: double space before 'timeout'

File: dali/base/dacsds.cpp [hunk 5]
```diff
@@ -1776,6 +1778,8 @@ IRemoteConnection *CClientSDSManager::connect(const char *xpath, SessionId id, u
             mb.read(connId);
             conn = new CRemoteConnection(*this, connId, xpath, id, mode & ~RTM_CREATE_MASK, timeout);
             assertex(conn);
+            if (queryProperties().getPropBool("Client/@LogConnection"))
+                DBGLOG("SDSManager::connect(): xpath<%s>, RemoteConnection ID<%" I64F "x>, mode<%x>,  timeout<%d>", xpath, connId, mode, timeout);
```
Comment (jakesmith) [inline dali/base/dacsds.cpp]: very trivial: double space before 'timeout'

File: esp/services/ws_smc/ws_smcService.cpp [hunk 6]
```diff
@@ -141,6 +141,9 @@ void CWsSMCEx::init(IPropertyTree *cfg, const char *process, const char *service
 
     xpath.setf("Software/EspProcess[@name=\"%s\"]/EspService[@name=\"%s\"]/ActivityInfoCacheSeconds", process, service);
     activityInfoCacheSeconds = cfg->getPropInt(xpath.str(), DEFAULTACTIVITYINFOCACHETIMEOUTSECOND);
+    xpath.setf("Software/EspProcess[@name=\"%s\"]/EspService[@name=\"%s\"]/LogDaliConnection", process, service);
+    if (cfg->getPropBool(xpath.str(), false))
```
Comment (jakesmith) [inline esp/services/ws_smc/ws_smcService.cpp]: trivial: don't really need ", false" .. getPropBool would return false by default anyway.

**Comment by jakesmith** [comment]
@wangkx - a couple of trivial comments, but not worth holding up the merge.

@richardkchapman - ready to merge.

**Comment by wangkx** [comment]
I will fix the 'trivial comments' and rebase into one commit.
